A few things

PurelyAndy opened the pull request:
Separate fog code into its own file
Match strength of enchantment glint to vanilla
Match fishing line and block selection to vanilla
Match mipmapping rules to vanilla (mostly - there is at least one remaining edge case that I don't know how to deal with.)

Specifically, vanilla has some strange rules for anisotropic filtering. It seems like all 2d blocks such as ladders, lily pads, etc. use filtered textures if viewed at a very oblique angle from a significant distance. This does not account for that, but, fortunately, it's a very minor discrepancy. Besides that, there may be some other full transparent blocks that are mipmapped besides leaves, but I haven't found them.

Bestsoft101 commented:
Thank you for the contribution, I have merged the fog and enchantment glint changes.

About the enchantment glint, I have glint strength set to 100% in game and because of that the previous code looked accurate, so instead of including the 0.75 multiplier I made it into an option.

I have not yet included the line rendering and mipmapping changes, because I don't really know what's going on in the code and I'd rather take some time to understand it first.

PurelyAndy commented:
> I have glint strength set to 100% in game and because of that the previous code looked accurate

I didn't even realize there was a setting for the enchantment glint, so that makes sense.

> I have not yet included the line rendering and mipmapping changes, because I don't really know what's going on in the code and I'd rather take some time to understand it first.

If it's helpful, I can try to explain why I changed what I changed here.

The mipmapping changes: In vanilla, most (all?) blocks with 2D transparent textures (ladders, vines, dried leaves, etc.) have mipmaps disabled. My changes account for that by checking if the vertex coordinates are at non-whole numbers, and if the highest-level mipmap meets a certain threshold for translucency. If so, it uses the normal texture. I also account for some exceptions to this heuristic. Looking at it closer, though, I may need to revisit that once I'm free to. I might be able to simplify that bit of code by quite a bit.

As for the line rendering, it's copied from https://github.com/shaderLABS/Base-150/blob/main/shaders/gbuffers_line.vsh and is (according to the comment) just a port of vanilla rendering to Iris shader code. I also changed the line width, because it didn't match vanilla. The difference, from what I can tell, is that in vanilla, the lines are allowed to sort of render through blocks slightly.

PurelyAndy commented:
It's a little more straightforward now:
Most cross-shaped blocks are never mipmapped
Most translucent blocks are never mipmapped
Exceptions to these are assigned 1.0 in block.properties
Some blocks that are neither translucent nor cross-shaped are also not mipmapped, those are assigned 2.0 in block.properties

Bestsoft101 commented:
Alright, the mipmapping fix https://github.com/Bestsoft101/Base-Shader/commit/d5cea7a640c57d54c1baa17a972ef47178487b53 and line rendering fix https://github.com/Bestsoft101/Base-Shader/commit/948445f9f1163c0e5bdb8e64086f372fa977d3e3 is in.

For the mipmapping fix I found a better solution using the cutout terrain shader, for all the blocks using the cutout mipped render layer, I made an exception by using block IDs.

The line shader is mostly the same, I didn't find much to improve there.

Bestsoft101 commented:
I guess I have to close this since I didn't merge the commits directly but the fixes are in:
https://github.com/Bestsoft101/Base-Shader/commit/3d4542ecd7e41e6db34c277284b4d30d3456cb23
https://github.com/Bestsoft101/Base-Shader/commit/d3dc204b01aff14a4c8cb79f3d67a38da7d4b35c
https://github.com/Bestsoft101/Base-Shader/commit/d5cea7a640c57d54c1baa17a972ef47178487b53
https://github.com/Bestsoft101/Base-Shader/commit/948445f9f1163c0e5bdb8e64086f372fa977d3e3

Thank you!
